tsdb/index/tsi1/partition.go

```diff
@@ -195,6 +195,7 @@ func (p *Partition) Open() error {
 		case LogFileExt:
 			f, err := p.openLogFile(filepath.Join(p.path, filename))
 			if err != nil {
+				Files(files).Close()
```

Comment:
I'd suggest:

```
once := sync.Once{}
defer func() {
  if (rErr != nil) {
    once.Do(Files(files).Close)
  }
}()

// filename loop

// 'Do' a no-op to cancel the 'once'
once.Do(func(){})
```

Reply:
Just had to make sure `NewFileSet` didn't mess up the `files` variable.  Looks good, using your suggestion.